core/src/test/scala/org/apache/spark/util/JsonProtocolSuite.scala

```diff
@@ -600,8 +611,16 @@ private[spark] object JsonProtocolSuite extends Assertions {
             assert(stageAttemptId1 === stageAttemptId2)
             assertSeqEquals[AccumulableInfo](updates1, updates2, (a, b) => a.equals(b))
           })
-        assertOptionEquals(e1.executorUpdates, e2.executorUpdates,
-        (e1: ExecutorMetrics, e2: ExecutorMetrics) => assertEquals(e1, e2))
+        assertSeqEquals[((Int, Int), ExecutorMetrics)](
+          e1.executorUpdates.toSeq.sortWith((x, y) => lexOrder(x._1, y._1)),
+          e2.executorUpdates.toSeq.sortWith((x, y) => lexOrder(x._1, y._1)),
```

Comment:
scala provides a default ordering on tuples, which is the same as your `lexOrder`, so I think you can get rid of that and this becomes  `.sortBy(_._1)`

Reply:
I see. Ok.